tests/PHPStan/Rules/Comparison/data/strict-comparison.php

```diff
@@ -982,3 +982,23 @@ function () {
 	INF !== INF;
 	NAN !== NAN;
 };
+
+class ArrayWithLongStrings
+{
+
+	public function doFoo()
+	{
+		$array = ['foofoofoofoofoofoofoo','foofoofoofoofoofoofob'];
+
+		foreach ($array as $value) {
+			if ('foofoofoofoofoofoofoo' === $value) {
+				echo 'nope';
+			} elseif ('foofoofoofoofoofoofob' === $value) {
+				echo 'nop nope';
+			} elseif (rand(0, 1) === 0) {
```

Comment:
this is a new variant of the existing test in the same file on line 654, with a additional else-if so we can see the error are reported when its not the last else-if